core/chains/evm/gas/universal_estimator_test.go

```diff
@@ -0,0 +1,557 @@
+
+		cfg := gas.UniversalEstimatorConfig{BlockHistorySize: 1}
+
+		u := gas.NewUniversalEstimator(logger.Test(t), client, cfg, chainID, nil)
+		_, err := u.FetchDynamicPrice()
+		assert.NoError(t, err)
+		dynamicFee, err := u.GetDynamicFee(tests.Context(t), maxPrice)
+		assert.NoError(t, err)
+		assert.Equal(t, maxPrice, dynamicFee.FeeCap)
+		assert.Equal(t, maxPrice, dynamicFee.TipCap)
+	})
+}
+
+func TestUniversalEstimatorBumpDynamicFee(t *testing.T) {
+	t.Parallel()
+
+	globalMaxPrice := assets.NewWeiI(100)
+	chainID := big.NewInt(0)
+
+	t.Run("bumps a previous attempt by BumpPercent", func(t *testing.T) {
+		client := mocks.NewUniversalEstimatorClient(t)
+		originalFee := gas.DynamicFee{
+			FeeCap: assets.NewWeiI(20),
+			TipCap: assets.NewWeiI(10),
+		}
+
+		// These values will be ignored because they are lower prices than the originalFee
+		feeHistoryResult := &ethereum.FeeHistory{
+			OldestBlock:  big.NewInt(1),
+			Reward:       [][]*big.Int{{big.NewInt(5), big.NewInt(50)}}, // first one represents market price and second one connectivity price
+			BaseFee:      []*big.Int{big.NewInt(5)},
+			GasUsedRatio: nil,
+		}
+		client.On("FeeHistory", mock.Anything, mock.Anything, mock.Anything).Return(feeHistoryResult, nil).Once()
+
+		cfg := gas.UniversalEstimatorConfig{
+			BlockHistorySize: 2,
+			BumpPercent:      50,
+			HasMempool:       true,
+		}
+
+		expectedFeeCap := originalFee.FeeCap.AddPercentage(cfg.BumpPercent)
+		expectedTipCap := originalFee.TipCap.AddPercentage(cfg.BumpPercent)
+
+		u := gas.NewUniversalEstimator(logger.Test(t), client, cfg, chainID, nil)
+		_, err := u.FetchDynamicPrice()
+		assert.NoError(t, err)
+		dynamicFee, err := u.BumpDynamicFee(tests.Context(t), originalFee, globalMaxPrice, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, expectedFeeCap, dynamicFee.FeeCap)
+		assert.Equal(t, expectedTipCap, dynamicFee.TipCap)
+	})
+
+	t.Run("fails if the original attempt is invalid", func(t *testing.T) {
+		client := mocks.NewUniversalEstimatorClient(t)
+		maxPrice := assets.NewWeiI(20)
+		cfg := gas.UniversalEstimatorConfig{BlockHistorySize: 1}
+
+		u := gas.NewUniversalEstimator(logger.Test(t), client, cfg, chainID, nil)
+		// nil original fee
+		var originalFee gas.DynamicFee
+		_, err := u.BumpDynamicFee(tests.Context(t), originalFee, maxPrice, nil)
+		assert.Error(t, err)
+
+		// tip cap is higher than fee cap
+		originalFee = gas.DynamicFee{
+			FeeCap: assets.NewWeiI(10),
+			TipCap: assets.NewWeiI(11),
+		}
+		_, err = u.BumpDynamicFee(tests.Context(t), originalFee, maxPrice, nil)
+		assert.Error(t, err)
+
+		// fee cap is equal or higher to max price
+		originalFee = gas.DynamicFee{
+			FeeCap: assets.NewWeiI(20),
+			TipCap: assets.NewWeiI(10),
+		}
+		_, err = u.BumpDynamicFee(tests.Context(t), originalFee, maxPrice, nil)
+		assert.Error(t, err)
+	})
+
+	t.Run("returns market prices bumped by BumpPercent if bumped original fee is lower", func(t *testing.T) {
```

Comment:
I think this description is incorrect based on the behavior. 
```suggestion
	t.Run("returns market prices if bumped original fee is lower", func(t *testing.T) {
```